modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -64,6 +67,9 @@
     public RustServerCodegen() {
         super();
 
+        // Force generating models for wrapping types
+        ModelUtils.setGenerateAliasAsModel(true);
```

Comment:
@richardwhiuk thanks for the PR. Do you mind explaining why we must enforce this option to true?

Some users simply want to reuse the same array definition through out the spec with aliases.

Reply:
Sorry, this should only apply to the Rust Generator - I need to figure out where to move it to so it's only invoked when the Rust generator is in use.

Currently, the generator assumes that the model will exist when generating the API bindings.

This means that it currently generates compile errors like:

```
error[E0412]: cannot find type `AnotherXmlArray` in module `models`
  --> output/openapi-v3/src/lib.rs:94:52
   |
94 |     fn xml_other_put(&self, string: Option<models::AnotherXmlArray>, context: &C) -> Box<Future<Item=XmlOtherPutResponse, Error=ApiError>>;
   |                                                    ^^^^^^^^^^^^^^^ did you mean `AnotherXmlInner`?
```

(for non aliasing, that should be `Option<Vec<....>>`.)

Worse, this is a backwards compatibility break - previously the generator did generate aliasing - #1729 changes this so the default is not to generate aliasing, meaning that the 4.x branch generates Rust code that doesn't compiler.

Longer term we can look at enhancing the Rust Server code generator to support generating code which compiles when aliasing is disabled (though unit structs in the manner that the Rust generator generates are fairly idiomatic, and allow for additional methods to be declared on them).

Reply:
> Sorry, this should only apply to the Rust Generator - I need to figure out where to move it to so it's only invoked when the Rust generator is in use.

Currently, it's already the case, which means only the Rust server generator will have this option enabled.

Thanks for the detailed explanation. We can certainly support generating array/map as model later on.

Reply:
> Currently, it's already the case, which means only the Rust server generator will have this option enabled.

I know what you mean now. I'll need to uncomment this line and show a warning message for the time being.

Reply:
Filed https://github.com/OpenAPITools/openapi-generator/pull/2714

Reply:
Yeah, I think the constructor must get run at startup, which means the other generators pick this up.

I'm slightly surprised we've defaulted `generateAliasAsModel` to false - this seems like the reverse of the default in 3.x?

Reply:
4.x is a branch with breaking changes (with or without fallback) and the main reason is that most generators do not support alias (array/map) as model so that's we set `generateAliasAsModel` to false as the default.